cqrs/src/main/scala/it/pagopa/interop/commons/cqrs/service/ReadModelService.scala

```diff
@@ -0,0 +1,73 @@
+package it.pagopa.interop.commons.cqrs.service
+
+import cats.implicits._
+import it.pagopa.interop.commons.cqrs.errors.ReadModelErrors.ReadModelMissingDataField
+import it.pagopa.interop.commons.cqrs.model.ReadModelConfig
+import it.pagopa.interop.commons.utils.TypeConversions._
+import org.mongodb.scala.MongoClient.DEFAULT_CODEC_REGISTRY
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.connection.NettyStreamFactoryFactory
+import org.mongodb.scala.{ConnectionString, Document, MongoClient, MongoClientSettings}
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final class ReadModelService(dbConfig: ReadModelConfig) {
+
+  private val client: MongoClient = MongoClient(
+    MongoClientSettings
+      .builder()
+      .applyConnectionString(new ConnectionString(dbConfig.connectionString))
+      .codecRegistry(DEFAULT_CODEC_REGISTRY)
+      .streamFactoryFactory(NettyStreamFactoryFactory())
+      .build()
+  )
+
+  private val db = client.getDatabase(dbConfig.dbName)
+
+  def findOne[T: JsonReader](collectionName: String, filter: Bson)(implicit ec: ExecutionContext): Future[Option[T]] =
+    find[T](collectionName, filter, offset = 0, limit = 1).map(_.headOption)
+
+  def find[T: JsonReader](collectionName: String, filter: Bson, offset: Int, limit: Int)(implicit
+    ec: ExecutionContext
+  ): Future[Seq[T]] =
+    // TODO Nice to have: remove the execution context. This works but I don't like throwing the error
```

Comment:
I do agree, but don't use cats traverse, use the stdlib api

Reply:
actually this is intended.
I want to avoid the request of 1 user eating all threads available in the service, causing disruption to other requests/users.
I prefer one request to use one thread in this case.

What do you think?

Reply:
That's an interesting topic. The usage depends on how the `.toFuture()` method works.
If it creates a collection of already-started futures (as I suspect and hope tbf), there's no significant difference since it changes just how the current thread awaits the group of futures. If it's not the case, `Future.traverse` will schedule all the futures together while cats' traverse will not. Let's go for the cats' traverse.